pkg/tfshim/sdk-v2/provider.go

```diff
@@ -161,11 +160,22 @@ func (p v2Provider) ReadDataDiff(
 	t string,
 	c shim.ResourceConfig,
 ) (shim.InstanceDiff, error) {
-	r, ok := p.tf.DataSourcesMap[t]
+	resource, ok := p.tf.DataSourcesMap[t]
 	if !ok {
 		return nil, fmt.Errorf("unknown resource %v", t)
 	}
-	diff, err := r.Diff(ctx, nil, configFromShim(c), p.tf.Meta())
+
+	providerOpts, err := getProviderOptions(p.opts)
+	if err != nil {
+		return nil, err
+	}
+	config := configFromShim(c)
+	rawConfig := makeResourceRawConfig(providerOpts.diffStrategy, config, resource)
```

Comment:
I'll follow up removing this diffStrategy in a separate PR.

Reply:
gotcha - we can then just bring the error fallback/warning logic to this method instead.